functionframeworks/frameworks.go

```diff
@@ -47,92 +50,102 @@ func registerHTTPFunction(path string, fn func(http.ResponseWriter, *http.Reques
 	return nil
 }
 
-func registerOpenFunction(fn func(*ofctx.OpenFunctionContext, interface{}) int, h *http.ServeMux) error {
+func registerOpenFunction(fn func(*ofctx.OpenFunctionContext, []byte) int, h *http.ServeMux) error {
 	ctx, err := ofctx.GetOpenFunctionContext()
 	if err != nil {
 		return err
 	}
 
-	if *ctx.Input.Enabled {
-		if ctx.Protocol == ofctx.HTTP {
-			h.HandleFunc(ctx.Input.Pattern, func(w http.ResponseWriter, r *http.Request) {
+	if ctx.Protocol == ofctx.HTTP {
+		if ctx.Runtime == ofctx.OpenFuncAsync {
+			openFuncAsyncServHandler = httpDaprd.NewService(fmt.Sprintf(":%s", ctx.Port))
+		} else {
+			h.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 				defer recoverPanicHTTP(w, "Function panic")
-				code := fn(ctx, r)
-				w.WriteHeader(code)
+				bodyBytes, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					w.WriteHeader(500)
+				} else {
+					code := fn(ctx, bodyBytes)
+					w.WriteHeader(code)
+				}
 			})
 		}
+	} else if ctx.Protocol == ofctx.GRPC {
```

Comment:
Is it necessary to determine whether the runtime is OpenFuncAsync?

Reply:
This is necessary. A judgment has been added to handle this case.